src/Middleware/HttpLogging/src/W3CLoggingMiddleware.cs

```diff
@@ -181,6 +183,17 @@ public async Task Invoke(HttpContext context)
                         shouldLog |= AddToList(elements, _cookieIndex, cookie.ToString());
                     }
                 }
+
+                if (_options.CurrentValue.AdditionalRequestHeaders != null)
+                {
+                    for (var i = 0; i < additionalHeadersLength; i++)
+                    {
+                        if (headers.TryGetValue(_options.CurrentValue.AdditionalRequestHeaders[i], out var headerValue))
+                        {
+                            shouldLog |= AddToList(elements, i + _fieldsLength, headerValue.ToString());
```

Comment:
What if the user passes an invalid header name? I think right now that invalid header name would wind up in the fields directive, then would be logged with a `-` on every line? My inclination is we should throw in that case, though I'm curious what @Tratcher @BrennanConroy @halter73 think

Reply:
Is that different from if they specified a header that the client never sends? I wouldn't worry about it.